openmc/mesh.py

```diff
@@ -274,6 +361,72 @@ def get_homogenized_materials(
 
         return homogenized_materials
 
+    def material_volumes(
+            self,
+            model: openmc.Model,
+            n_samples: int | tuple[int, int, int] = 10_000,
+            max_materials: int = 4,
+            **kwargs
+    ) -> MeshMaterialVolumes:
+        """Determine volume of materials in each mesh element.
+
+        This method works by raytracing repeatedly through the mesh to count the
+        estimated volume of each material in all mesh elements. Three sets of
+        rays are used: one set parallel to the x-axis, one parallel to the
+        y-axis, and one parallel to the z-axis.
+
+        .. versionadded:: 0.15.1
+
+        Parameters
+        ----------
+        model : openmc.Model
+            Model containing materials.
+        n_samples : int or 3-tuple of int
+            Total number of rays to sample. The number of rays in each direction
+            is determined by the aspect ratio of the mesh bounding box. When
+            specified as a 3-tuple, it is interpreted as the number of rays in
+            the x, y, and z dimensions.
+        max_materials : int, optional
+            Estimated maximum number of materials in any given mesh element.
+        **kwargs : dict
+            Keyword arguments passed to :func:`openmc.lib.init`
+
+        Returns
+        -------
+        Dictionary-like object that maps material IDs to an array of volumes
+        equal in size to the number of mesh elements.
+
+        """
+        import openmc.lib
+
+        with change_directory(tmpdir=True):
+            # In order to get mesh into model, we temporarily replace the
+            # tallies with a single mesh tally using the current mesh
+            original_tallies = model.tallies
+            new_tally = openmc.Tally()
+            new_tally.filters = [openmc.MeshFilter(self)]
+            new_tally.scores = ['flux']
+            model.tallies = [new_tally]
+
+            # Export model to XML
+            model.export_to_model_xml()
+
+            # Get material volume fractions
+            kwargs.setdefault('output', True)
```

Comment:
I noted that this default was different than the value of the `openmc.lib` method so I made them consistent in my latest commit here, 8a47fbdb4c34599bcd27df2afeae5d160230001f. I thought `True` was sensible for this kind of operation as the operation can still take a while with a lot of rays and it's probably good to see that OpenMC is doing something when calling this from Python. 

But I also understand wanting to limit output as much as possible. Main thing is that the values are consistent I'd say.